### PR TITLE
Ability Popup Fixes + Longer Ability Names config

### PR DESCRIPTION
**Pull request by ghoulslash**
- Make ability popup larger and fix printing ability names
- add a function to clean the ability name to fix wandering spirit
- Config option, `B_EXPANDED_ABILITY_NAMES` to increase ability name length from 12 to 16 chars (and associated ability name fixes
pokeemerald-1
)

### CODE REVIEW

File: src/battle_interface.c [hunk 1]
```diff
@@ -2976,111 +2987,125 @@ static void PrintAbilityOnAbilityPopUp(u32 ability, u8 spriteId1, u8 spriteId2)
...
+				pixelCount--;
+			}
+			i++;
+		}
+	}
+	else
+	{
+		for (i = 0; i < pixelCount / 2; i++)
+			dest[i] = src[i];
+	}
```
Comment (LOuroboros) [inline src/battle_interface.c:3090]: Spaces instead of tabs.

File: src/battle_interface.c [hunk 2]
```diff
@@ -2976,111 +2987,125 @@ static void PrintAbilityOnAbilityPopUp(u32 ability, u8 spriteId1, u8 spriteId2)
...
 }
 
 static void RestoreOverwrittenPixels(u8 *tiles)
 {
-    u32 i;
-    u8 *buffer = Alloc(sizeof(sAbilityPopUpGfx) * 2);
+	u32 i;
+	u8 *buffer = Alloc(sizeof(sAbilityPopUpGfx) * 2);
```
Comment (LOuroboros) [inline src/battle_interface.c:3096]: Spaces instead of tabs.

File: src/battle_interface.c [hunk 3]
```diff
@@ -2976,111 +2987,125 @@ static void PrintAbilityOnAbilityPopUp(u32 ability, u8 spriteId1, u8 spriteId2)
...
 
-    CpuCopy32(tiles, buffer, sizeof(sAbilityPopUpGfx));
+	CpuCopy32(tiles, buffer, sizeof(sAbilityPopUpGfx));
```
Comment (LOuroboros) [inline src/battle_interface.c:3098]: Spaces instead of tabs.

File: src/battle_interface.c [hunk 4]
```diff
@@ -2976,111 +2987,125 @@ static void PrintAbilityOnAbilityPopUp(u32 ability, u8 spriteId1, u8 spriteId2)
...
-        CopyPixels(buffer + sOverwrittenPixelsTable[i][0],
-                   sAbilityPopUpGfx + sOverwrittenPixelsTable[i][0],
-                   sOverwrittenPixelsTable[i][1]);
-    }
+	for (i = 0; i < ARRAY_COUNT(sOverwrittenPixelsTable); i++)
+	{
+		CopyPixels(buffer + sOverwrittenPixelsTable[i][0],
+				   sAbilityPopUpGfx + sOverwrittenPixelsTable[i][0],
+				   sOverwrittenPixelsTable[i][1]);
+	}
```
Comment (LOuroboros) [inline src/battle_interface.c:3105]: Spaces instead of tabs.

File: src/battle_interface.c [hunk 5]
```diff
@@ -2976,111 +2987,125 @@ static void PrintAbilityOnAbilityPopUp(u32 ability, u8 spriteId1, u8 spriteId2)
...
 
-    CpuCopy32(buffer, tiles, sizeof(sAbilityPopUpGfx));
-    Free(buffer);
+	CpuCopy32(buffer, tiles, sizeof(sAbilityPopUpGfx));
+	Free(buffer);
```
Comment (LOuroboros) [inline src/battle_interface.c:3108]: Spaces instead of tabs.

File: include/battle_main.h [hunk 6]
```diff
@@ -24,7 +24,12 @@ struct UnknownPokemonStruct4
 struct ChooseMoveStruct;
 
 #define TYPE_NAME_LENGTH 6
+
+#if B_EXPANDED_ABILITY_NAMES
+#define ABILITY_NAME_LENGTH 16
+#else
 #define ABILITY_NAME_LENGTH 12
+#endif
```
Comment (LOuroboros) [inline include/battle_main.h:32]:
```suggestion
#endif
```

File: src/battle_interface.c [hunk 7]
```diff
@@ -2855,83 +2855,94 @@ static const struct SpriteTemplate sSpriteTemplate_AbilityPopUp2 =
 static const s16 sAbilityPopUpCoordsDoubles[MAX_BATTLERS_COUNT][2] =
 {
     {29, 80}, // player left
-    {204, 19}, // opponent left
-    {29, 97}, // player right
-    {204, 36}, // opponent right
+	{186, 19}, // opponent left
+	{29, 97}, // player right
+	{186, 36}, // opponent right
```
Comment (LOuroboros) [inline src/battle_interface.c:2860]: Spaces instead of tabs.

File: src/battle_interface.c [hunk 8]
```diff
@@ -2855,83 +2855,94 @@ static const struct SpriteTemplate sSpriteTemplate_AbilityPopUp2 =
...
 };
 
 static const s16 sAbilityPopUpCoordsSingles[MAX_BATTLERS_COUNT][2] =
 {
-    {29, 93}, // player
-    {204, 23}, // opponent
+    {29, 97}, // player
+	{186, 57}, // opponent
```
Comment (LOuroboros) [inline src/battle_interface.c:2866]: Spaces instead of tabs.

File: src/battle_interface.c [hunk 9]
```diff
@@ -2855,83 +2855,94 @@ static const struct SpriteTemplate sSpriteTemplate_AbilityPopUp2 =
...
 static u8* AddTextPrinterAndCreateWindowOnAbilityPopUp(const u8 *str, u32 x, u32 y, u32 color1, u32 color2, u32 color3, u32 *windowId)
 {
-    u8 color[3] = {color1, color2, color3};
-    struct WindowTemplate winTemplate = {0};
-    winTemplate.width = 8;
-    winTemplate.height = 2;
+	u8 color[3] = {color1, color2, color3};
+	struct WindowTemplate winTemplate = {0};
+	winTemplate.width = 8;
+	winTemplate.height = 2;
```
Comment (LOuroboros) [inline src/battle_interface.c:2874]: Spaces instead of tabs.

File: src/battle_interface.c [hunk 10]
```diff
@@ -2855,83 +2855,94 @@ static const struct SpriteTemplate sSpriteTemplate_AbilityPopUp2 =
...
 
-    *windowId = AddWindow(&winTemplate);
-    FillWindowPixelBuffer(*windowId, (color1 << 4) | (color1));
+	*windowId = AddWindow(&winTemplate);
+	FillWindowPixelBuffer(*windowId, PIXEL_FILL(color1));
```
Comment (LOuroboros) [inline src/battle_interface.c:2877]: Spaces instead of tabs.

File: src/battle_interface.c [hunk 11]
```diff
@@ -2855,83 +2855,94 @@ static const struct SpriteTemplate sSpriteTemplate_AbilityPopUp2 =
...
 
-    AddTextPrinterParameterized4(*windowId, 0, x, y, 0, 0, color, -1, str);
-    return (u8*)(GetWindowAttribute(*windowId, WINDOW_TILE_DATA));
+	AddTextPrinterParameterized4(*windowId, 0, x, y, 0, 0, color, -1, str);
+	return (u8*)(GetWindowAttribute(*windowId, WINDOW_TILE_DATA));
```
Comment (LOuroboros) [inline src/battle_interface.c:2880]: Spaces instead of tabs.

File: src/battle_interface.c [hunk 12]
```diff
@@ -2855,83 +2855,94 @@ static const struct SpriteTemplate sSpriteTemplate_AbilityPopUp2 =
...
+		do
+		{
+			if (arg3)
+				CpuCopy32(windowTileData + 16, dest + 16, 16);
+			else
+				CpuCopy32(windowTileData + 20, dest + 20, 12);
+			dest += 32, windowTileData += 32;
+			arg2--;
+		} while (arg2 != 0);
+	}
```
Comment (LOuroboros) [inline src/battle_interface.c:2897]: Spaces instead of tabs.

File: src/battle_interface.c [hunk 13]
```diff
@@ -2855,83 +2855,94 @@ static const struct SpriteTemplate sSpriteTemplate_AbilityPopUp2 =
...
+			text2[i] = str[MAX_CHARS_PRINTED + i];
+			if (text2[i] == EOS)
+				break;
+		}
+		text2[i] = EOS;
+
+		windowTileData = AddTextPrinterAndCreateWindowOnAbilityPopUp(text2, x2, y, color1, color2, color3, &windowId);
+		TextIntoAbilityPopUp(spriteTileData2, windowTileData, 3, (y == 0));
+		RemoveWindow(windowId);
+	}
```
Comment (LOuroboros) [inline src/battle_interface.c:2934]: Spaces instead of tabs.

File: src/battle_interface.c [hunk 14]
```diff
@@ -2976,111 +2987,125 @@ static void PrintAbilityOnAbilityPopUp(u32 ability, u8 spriteId1, u8 spriteId2)
...
-+ ((((y) - ((y / 8) * 8))) * 4)                    \
-/*Add pixels by X*/                                \
+#define PIXEL_COORDS_TO_OFFSET(x, y)(			\
+/*Add tiles by X*/								\
+((y / 8) * 32 * 8)								\
+/*Add tiles by X*/								\
++ ((x / 8) * 32)								\
+/*Add pixels by Y*/								\
++ ((((y) - ((y / 8) * 8))) * 4)					\
+/*Add pixels by X*/								\
```
Comment (LOuroboros) [inline src/battle_interface.c:3002]: Spaces instead of tabs.

File: src/battle_interface.c [hunk 15]
```diff
@@ -2976,111 +2987,125 @@ static void PrintAbilityOnAbilityPopUp(u32 ability, u8 spriteId1, u8 spriteId2)
...
+	{PIXEL_COORDS_TO_OFFSET(0, 47), 8},
+	//{PIXEL_COORDS_TO_OFFSET(0, 48), 8},   // cuts off the top of the 'G' in Neutralizing Gas
+	{PIXEL_COORDS_TO_OFFSET(8, 45), 8},
+	{PIXEL_COORDS_TO_OFFSET(8, 46), 8},
+	{PIXEL_COORDS_TO_OFFSET(8, 47), 8},
+	{PIXEL_COORDS_TO_OFFSET(8, 48), 8},
+	{PIXEL_COORDS_TO_OFFSET(16, 45), 8},
+	{PIXEL_COORDS_TO_OFFSET(16, 46), 8},
+	{PIXEL_COORDS_TO_OFFSET(16, 47), 8},
+	{PIXEL_COORDS_TO_OFFSET(16, 48), 8},
```
Comment (LOuroboros) [inline src/battle_interface.c:3064]: Spaces instead of tabs.

**Review by LOuroboros — CHANGES_REQUESTED** [review]

**Comment by Jaizu** [comment]
Shouldn't those edits to the gen 3 ability names have ifdefs?

**Comment by ghoulslash** [comment]
> Shouldn't those edits to the gen 3 ability names have ifdefs?

Its one big ifdef

**Comment by ghoulslash** [comment]
formatting fixed. ready for review

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**